File: src/programmers/helpers/IExpressionEntry.ts
import type ts from "typescript";

import type { Metadata } from "../../schemas/metadata/Metadata";

export interface IExpressionEntry<
  Expression extends ts.ConciseBody = ts.ConciseBody,
> {
  input: ts.Expression;
  key: Metadata;
  meta: Metadata;
  expression: Expression;
}

File: packages/typia/native/core/programmers/helpers/IExpressionEntry.go
package helpers

import (
  shimast "github.com/microsoft/typescript-go/shim/ast"
  nativemetadata "github.com/samchon/typia/packages/typia/native/core/schemas/metadata"
)

type IExpressionEntry struct {
  Input      *shimast.Node
  Key        *nativemetadata.MetadataSchema
  Meta       *nativemetadata.MetadataSchema
  Expression *shimast.Node
}
